suprb/rule/initialization.py

```diff
@@ -19,25 +20,42 @@ class RuleInit(BaseComponent, metaclass=ABCMeta):
     Parameters
         ----------
         bounds: np.ndarray
-            The generated interval will lie within the absolute bounds.
+            The absolute bounds of the feature space.
         model: RegressorMixin
             Local model used for fitting the intervals.
     """
 
-    def __init__(self, bounds: np.ndarray = None, model: RegressorMixin = None, fitness: RuleFitness = None):
+    def __init__(self, bounds: np.ndarray = None,
+                 model: RegressorMixin = None,
+                 fitness: RuleFitness = None,
+                 matching_type: MatchingFunction = None):
         self.bounds = bounds
         self.model = model
         self.fitness = fitness
+        self.matching_type = matching_type
 
-        self._validate_components(model=Ridge(alpha=0.01), fitness=VolumeWu())
+        self._validate_components(model=Ridge(alpha=0.01),
+                                  fitness=VolumeWu())
+
+    @property
+    def matching_type(self):
+        return self._matching_type
+
+    @matching_type.setter
+    def matching_type(self, matching_type):
+        self._matching_type = matching_type
+        if isinstance(self.matching_type, OrderedBound):
+            self.generate_matchf = self.ordered_bound
```

Comment:
Can we use a more descriptive name for `generate_matchf`? If not then `generate_match_function` would do

Reply:
I did originally shorten it to make the code less convoluted but as it stands it is only ever used here so `generate_match_function` should be fine. Can't think of anything more descriptive than that for a method that generates a new instance of a matching function